optional-integration-analysis/merge-sce.R

```diff
@@ -0,0 +1,108 @@
+## This script takes in individual SCE objects and merges them into one SCE object
+## for a data integration analysis.
+
+## Set up ----------------------------------------------------------------------
+
+# Load libraries
+library(optparse)
+library(SingleCellExperiment)
+
+# Declare command line options
+option_list <- list(
+  make_option(
+    opt_str = c("--input_metadata_tsv"),
+    type = "character",
+    help = "Path to the TSV file containing the list of library IDs corresponding to the libraries being integrated, must have `library_id` and `processed_sce_filepath` columns."
+  ),
+  make_option(
+    opt_str = c("-o", "--output_sce_file"),
+    type = "character",
+    help = "Path to output RDS file, must end in .rds"
+  ),
+  optparse::make_option(
+    c("--project_root"),
+    type = "character",
+    default = NULL,
+    help = "Path to the root directory for the R project and where the `utils` folder lives."
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 1,
+    help = "Number of multiprocessing threads to use"
+  )
+)
+
+# Read the arguments passed
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# If project root is not provided use here::here()
+if(is.null(opt$project_root)){
+  project_root <- here::here()
+} else {
+  project_root <- opt$project_root
+}
+
+# Source in set up functions
+source(file.path(project_root, "utils", "setup-functions.R"))
+
+# Check R version
+check_r_version()
+
+# Set up renv
+setup_renv(project_filepath = project_root)
+
+# Check that file extension for output file is correct
+if(!(stringr::str_ends(opt$output_sce_file, ".rds"))){
+  stop("output file name must end in .rds")
+}
+
+# Check that input metadata file is provided
+if(is.null(opt$input_metadata_tsv)){
+  stop("TSV file with the list of library IDs associated with the individual SCE objects to merge is missing.")
+} else {
+  # List of library ids
+  input_metadata <- readr::read_tsv(opt$input_metadata_tsv)
+}
+
+# List of SCE filepaths
+sce_files <- input_metadata$processed_sce_filepath
```

Comment:
```suggestion
sce_files <- input_metadata$processed_sce_filepath |> 
purrr::set_names(input_metadata$library_id)
```
I think you want to set the names of the `sce_files` here to be the `library_id` column. Otherwise the `merge_sce_list` function will print out a warning and not properly identify which library belongs to which SCE object. Also if you name the files here, when you read in the RDS files using `purrr` they should retain the names.